pep-9999.rst

```diff
@@ -304,11 +305,103 @@ Security Implications
+consists of:
+
+1. Understanding the nature of their dependency. Do they only need runtime 
+   libraries, or do they need development packages for build-time concerns?
+   This understanding feeds into PEP 725, which specifies the expression of
+   these dependencies in metadata.
+2. Looking up the PURL. This can either mean knowing the name of the package 
+   in their package ecosystem, and then inverse-mapping that to the PURL, or
+   it can mean looking up the PURL directly.
+3. When a package maintainer does not find an appropriate mapping, they should look 
+   for a fitting PURL. It can be the case that although a PURL is registered, not every 
+   package ecosystem has a corresponding mapping. If no appropriate PURL exists,
+   the package maintainer may consider submitting a new PURL to the central registry.
+
+An overall workflow diagram might look like this:
+
+.. mermaid::
+
+   flowchart TD
+      A[Python package author with new external dependency] --> |Looks in| B(PURL/description collection)
+      B --> | Find PURL OK | E(Add PURL to pyproject.toml)
+      A --> | Looks in | C(Ecosystem mapping file)
+      C --> | Finds familiar ecosystem package name | D(Inverse map ecosystem package name to PURL)
+      D --> | Mapping exists | E
+      B --> | PURL not found | F(Submit PURL proposal to PURL/description collection)
+      F --> | Accepted | G(Mapping maintainers notified of missing PURL mappings)
+      D --> | Mapping missing. User looks in PURL collection. | B
+      B --> | Was mapping missing? | H(User may contribute entry to mapping)
+
+Package ecosystem maintainers usage
+-----------------------------------
+
+Any packages that express a PURL dependency that does not have a mapping in a given package
+ecosystem will mean a failed installation. It is thus critically important that each package
+ecosystem maintain their mappings. Key to this will be automation. Some ideas for automation
+are:
+
+1. Alert mapping maintainers whenever a new PURL is added to the registry (probably noisy).
+2. Provide tools that allow maintainers to diff their mappings to the registry contents to
+   quickly identify missing entries.
+3. Provide automated tooling that submits PRs to known mapping locations, such that maintainers
+   need only fill in the ecosystem package name.
+4. Provide status for each PURL, to readily identify which PURLs need attention.
+
+This maintenance is likely to be a lot of work to establish the initial mapping, but ideally small
+on an ongoing basis.
+
+
+Central PURL registry maintainers
+---------------------------------
+
+Central PURL registry maintainers curate the collection of PURLs. These contributors
+need to be able to refer to clearly defined rules for when a new PURL can be defined. It is
+undesirable to be loose with PURL definition, because each definition implies maintenance in
+the mappings in many other places.
+
+
+End user package consumers
```

Comment:
I'd aim for: "no change in user experience by default, and end users do not need to know about this mechanism unless they opt in (which they may want to do to, for example, reduce their bandwidth and disk space usage".